metadata-ingestion/src/datahub/ingestion/source/looker/looker_source.py

```diff
@@ -1104,6 +1122,80 @@ def get_workunit_processors(self) -> List[Optional[MetadataWorkUnitProcessor]]:
             ).workunit_processor,
         ]
 
+    def emit_independent_looks_mcp(
+        self, dashboard_element: LookerDashboardElement
+    ) -> Iterable[MetadataWorkUnit]:
+        yield from [
+            MetadataWorkUnit(id=f"looker-{mce.proposedSnapshot.urn}", mce=mce)
+            if isinstance(mce, MetadataChangeEvent)
+            else MetadataWorkUnit(
+                id=f"looker-{mce.aspectName}-{mce.entityUrn}", mcp=mce
+            )
+            for mce in self._make_chart_metadata_events(
+                dashboard_element=dashboard_element, dashboard=None
+            )
+        ]
+
+        mcp: MetadataChangeProposalWrapper = self._make_metrics_dimensions_chart_mcp(
+            dashboard_element
+        )
+        yield MetadataWorkUnit(id=f"looker-{mcp.entityUrn}", mcp=mcp)
+
+    def extract_independent_looks(self) -> Iterable[MetadataWorkUnit]:
+        """
+        Emit MetadataWorkUnit for looks which are not part of any Dashboard
+        """
+        if self.source_config.extract_independent_looks is False:
+            return
+
+        logger.debug("Extracting looks not part of Dashboard")
+        look_fields: List[str] = [
+            "id",
+            "title",
+            "description",
+            "query_id",
+        ]
+        query_fields: List[str] = [
+            "id",
+            "view",
+            "model",
+            "dynamic_fields",
+            "filters",
+            "fields",
+            "slug",
+        ]
+
+        all_looks: List[Look] = self.looker_api.all_looks(fields=look_fields)
+        for look in all_looks:
+            if look.id in self.reachable_look_registry:
+                # This look is reachable from Dashboard
+                continue
+
+            if look.query_id is None:
+                logger.info(f"query_id is None for look {look.title}({look.id})")
+                continue
+
+            query: Query = self.looker_api.get_query(look.query_id, query_fields)
+
+            dashboard_element: Optional[
+                LookerDashboardElement
+            ] = self._get_looker_dashboard_element(
+                DashboardElement(
+                    id=f"looks_{look.id}",  # to avoid conflict with element.id prefixing "looks_" to look.id.
```

Comment:
```suggestion
                    id=f"looks_{look.id}",  # to avoid conflict with non-standalone looks (element.id prefixes), we add the "looks_" prefix to look.id.
```

Reply:
done